### DISCUSSION TITLE
Question: Maybe untangle Borsh into it's own package?

### DISCUSSION THREAD

**Discussion by neverything** [verze-app/solana-php-sdk#41] [Ideas]
**What would this mean?**

- Should the Buffer become part of the Borsh package?
- Would the Exceptions become part of the Borsh package?
- What about the PublicKey that now is part as of this change (https://github.com/tighten/solana-php-sdk/commit/d032405c5f5ef7d91d7f245a2e0f11cea6eb3c4f#diff-bc3a24552d6e6304731de9cd989008ec466e662dfd2d77273ea76f4749755970)?
- Anything I'm missing?

Would love to help out

**Comment by neverything** [comment]
Ping @mattstauffer, only if you have a moment good ser

**Comment by mattstauffer** [comment]
Whoops, I'm sorry I didn't see this originally! I would love to extract it... but I don't know how to answer your questions. I think exceptions are the easiest: we can just make custom exceptions in that package and update this package to be aware of them.

With Buffer, if we need to use buffers in our Borsh package, then this package can just consume those, so that's fine.

I think the same will have to be true about the `PublicKey`. Basically anything the Borsh package needs, should move there, and then this package can just get them from there, even if that seems a bit silly.

**Comment by neverything** [comment]
Maybe a bit of additional background. Since there are a bunch of implementations for [Borsh](https://github.com/near/borsh#implementations), the question came up if it makes sense to create a standalone implementation for PHP.

Let's discuss this
